R/conditions.R

```diff
@@ -25,15 +22,30 @@ warn_on_lossy_cast <- function(expr, x_ptype = NULL, to_ptype = NULL) {
 
 #' @export
 cnd_header.bignum_warning_cast_lossy <- function(cnd, ...) {
-  x_label <- format_arg_label(vec_ptype_full(cnd$x), cnd$x_arg)
-  to_label <- format_arg_label(vec_ptype_full(cnd$to), cnd$to_arg)
-  loss_type <- loss_type(cnd$loss_type)
+  x_label <- format_arg_label(vec_ptype_full(cnd$error$x), cnd$error$x_arg)
+  to_label <- format_arg_label(vec_ptype_full(cnd$error$to), cnd$error$to_arg)
+  loss_type <- loss_type(cnd$error$loss_type)
   paste0(
     "Loss of ", loss_type, " while converting from ",
     x_label, " to ", to_label, "."
   )
 }
 
+#' @export
+cnd_body.bignum_warning_cast_lossy <- function(cnd, ...) {
+  cnd_body(cnd$error)
+}
+
+#' @export
+cnd_footer.bignum_warning_cast_lossy <- function(cnd, ...) {
+  cnd_footer(cnd$error)
+}
+
+#' @export
+conditionMessage.bignum_warning_cast_lossy <- function(c) {
```

Comment:
Can you make this `cnd_header()`?

Starting from rlang 1.0, it will be UB to overwrite `conditionMessage()` for conditions thrown with rlang.

Reply:
Thanks for the review!

In rlang 0.4.11, there isn't a `conditionMessage()` method for warnings thrown by `warn()` because the `"rlang_warning"` class doesn't exist yet. I need to export this method for the warning message to be shown. Once rlang 1.0.0 is released, I can remove the method completely and rely on `conditionMessage.rlang_warning()`.

I could add a `# TODO: remove when rlang 1.0 is released` comment here. Or is there a better approach to handling the rlang 1.0 release that I'm not recognizing?

Reply:
Ah good point, removing the method in the future sounds good.